packages/editor/src/components/provider/use-post-editor-layout.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+import { useSettings } from '@wordpress/block-editor';
+
+export function usePostEditorLayout( templatePostContentBlock ) {
+	const [ globalStylesLayout ] = useSettings( 'layout' );
+	const { layout: templateLayoutDefinition } =
+		templatePostContentBlock?.attributes ?? {};
+	const layout = useMemo( () => {
+		if ( ! templateLayoutDefinition ) {
+			return globalStylesLayout;
+		}
+		return templateLayoutDefinition &&
+			( templateLayoutDefinition?.type === 'constrained' ||
+				templateLayoutDefinition?.inherit ||
+				templateLayoutDefinition?.contentSize ||
+				templateLayoutDefinition?.wideSize )
+			? {
+					...globalStylesLayout,
+					...templateLayoutDefinition,
+					type: 'constrained',
+			  }
+			: { ...globalStylesLayout, ...layout, type: 'default' };
```

Comment:
```suggestion
			: {
					...globalStylesLayout,
					...templateLayoutDefinition,
					type: 'default',
			  };
```
I think this should fix the error @andrewserong [noticed](https://github.com/WordPress/gutenberg/pull/56542#pullrequestreview-1751670873).

I think you meant to destructure `templateLayoutDefinition` as `layout` is already set as the output of `useMemo`?

Reply:
Great catch. Thank you!